src/main/java/core/basesyntax/servise/ShopServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.servise;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.servise.operation.OperationHandler;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Map<String, Integer> storage = new HashMap<>(); // Хранилище фруктов
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.get(transaction.getOperation());
+            String fruit = transaction.getFruit();
+            int quantity = transaction.getQuantity();
+
+            handler.apply(fruit, quantity, storage);
+        }
+    }
+
+    public Map<String, Integer> getStorage() {
```

Comment:
The encapsulation principle is not followed properly here. The storage map should not be exposed directly, as it allows for its contents to be modified from outside the class. Consider providing a defensive copy or an unmodifiable view of the map.